src/components/onboarding/Onboarding.tsx

```diff
@@ -17,60 +18,101 @@ const Onboarding: React.FC = () => {
   const ONBOARDING_KEY = "onboardingComplete";
   const [targetRect, setTargetRect] = useState<DOMRect | null>(null);
   const rafRef = useRef<number | null>(null);
+  const { isAdmin, isDeveloper, isTester, isAccountant } = useUserRole();
 
-  // Always get the latest steps for the current language
+  /**
+   * Whether the current user has permission to view onboarding.
```

Comment:
As a general rule we only use TSDOcs for function comments. But this is fine here, typically if we wanted to comment something other than a function we would use // syntax